### PR TITLE
Changes to support Bucket DNS

### PR DESCRIPTION
**Pull request by kerneltime**
1. Use standard 443 and 80 ports
2. Changes to support Bucket DNS.

Changes tested via curl for now.

TODO:

- [x] Minio side changes

- [x] MC changes

- [x] Testing with mc

### CODE REVIEW

**Review by nitisht — APPROVED** [review]
LGTM

**Review by harshavardhana — DISMISSED** [review]

**Comment by harshavardhana** [comment]
Please look at the conflicts @kerneltime

**Review by harshavardhana — APPROVED** [review]

File: pkg/resources/services/service.go [hunk 1]
```diff
@@ -20,13 +20,27 @@ package services
...
+		port = 443
+		name = miniov1.MinIOServiceHTTPSPortName
+	} else {
+		port = 80
+		name = miniov1.MinIOServiceHTTPPortName
+	}
+	minioPort := corev1.ServicePort{
+		Port:       port,
+		Name:       name,
+		TargetPort: intstr.FromInt(miniov1.MinIOPort),
```
Comment (harshavardhana) [inline pkg/resources/services/service.go:42]: why not use strconv.Itoa() ?

File: pkg/resources/services/service.go [hunk 2]
```diff
@@ -46,7 +60,20 @@ func NewClusterIPForMinIO(t *miniov1.Tenant) *corev1.Service {
...
```
Comment (harshavardhana) [inline pkg/resources/services/service.go:75]: why not use strconv.Itoa()?

**Review by harshavardhana — CHANGES_REQUESTED** [review]

File: pkg/resources/services/service.go (hunk 1 above)
Comment (kerneltime) [reply pkg/resources/services/service.go:42]: Using the API framework to keep the types the same: https://godoc.org/k8s.io/api/core/v1#ServicePort

File: pkg/resources/services/service.go (hunk 2 above)
Comment (kerneltime) [reply pkg/resources/services/service.go:75]: Using the API framework to keep the types the same: https://godoc.org/k8s.io/api/core/v1#ServicePort

File: pkg/resources/services/service.go (hunk 1 above)
Comment (harshavardhana) [reply pkg/resources/services/service.go:42]: Wow, that is so ugly!

**Review by harshavardhana — APPROVED** [review]

**Review by nitisht — APPROVED** [review]

**Pull request merged**